File: backend/app/schemas/doctor_assistant.py
from typing import Literal, Optional

from pydantic import BaseModel, Field


ContextType = Literal["patient", "today_reports", "reports_range", "general"]
Priority = Literal["high", "medium", "low"]
Status = Literal["success", "error"]
TaskType = Literal[
    "answer",
    "triage_queue",
    "draft_referral_letter",
    "draft_followup_plan",
]
TimeframePreset = Literal["today", "7d", "30d", "90d", "all", "custom"]


class DoctorAssistantExplainRequest(BaseModel):
    patient_id: Optional[int] = Field(default=None, description="Optional patient id for patient context")
    doctor_query: str = Field(min_length=1, max_length=4000)
    context_type: ContextType = Field(default="general")
    task: TaskType = Field(default="answer", description="What you want the assistant to do")

    # Optional time filtering (used for today_reports/reports_range, and optionally patient history)
    timeframe: Optional[TimeframePreset] = Field(default=None, description="Quick time filter preset")
    start_date: Optional[str] = Field(default=None, description="YYYY-MM-DD (required if timeframe=custom)")
    end_date: Optional[str] = Field(default=None, description="YYYY-MM-DD (required if timeframe=custom)")


class DoctorAssistantExplainResponse(BaseModel):
    status: Status
    answer: Optional[str] = None
    priority: Optional[Priority] = None
    summary: Optional[str] = None

    # error payload (when status=error)
    message: Optional[str] = None
    fallback: Optional[str] = None

    # additional structured metadata
    priority_reason: Optional[str] = None
    worsening_detected: Optional[bool] = None
    provider_used: Optional[str] = None

File: frontend1/src/services/doctorAssistant.ts
import api from "./api";

export type DoctorAssistantContextType = "patient" | "today_reports" | "reports_range" | "general";
export type DoctorAssistantPriority = "high" | "medium" | "low";
export type DoctorAssistantTask = "answer" | "triage_queue" | "draft_referral_letter" | "draft_followup_plan";
export type DoctorAssistantTimeframe = "today" | "7d" | "30d" | "90d" | "all" | "custom";

export interface DoctorAssistantExplainRequest {
  patient_id?: number | null;
  doctor_query: string;
  context_type: DoctorAssistantContextType;
  task?: DoctorAssistantTask;
  timeframe?: DoctorAssistantTimeframe | null;
  start_date?: string | null; // YYYY-MM-DD
  end_date?: string | null; // YYYY-MM-DD
}

export interface DoctorAssistantExplainResponse {
  status: "success" | "error";
  answer?: string | null;
  priority?: DoctorAssistantPriority | null;
  summary?: string | null;

  message?: string | null;
  fallback?: string | null;

  priority_reason?: string | null;
  worsening_detected?: boolean | null;
  provider_used?: string | null;
}

export async function explainDoctorQuery(payload: DoctorAssistantExplainRequest) {
  const { data } = await api.post<DoctorAssistantExplainResponse>("/doctor-assistant/explain", payload);
  return data;
}
